ui/cryptomaterial/dropdown.go

```diff
@@ -134,69 +183,127 @@ func (d *DropDown) Changed() bool {
+		Clickable: clickable,
+		Padding:   layout.UniformInset(padding),
+		Border:    Border{Radius: radius},
+	}.Layout(gtx,
+		layout.Rigid(func(gtx C) D {
+			if item.Icon == nil {
+				return D{}
+			}
+
+			return item.Icon.Layout24dp(gtx)
+		}),
+		layout.Flexed(1, func(gtx C) D {
+			if item.DisplayFn != nil {
+				return item.DisplayFn(gtx)
+			}
+
+			ll := LinearLayout{
+				Width:     MatchParent,
+				Height:    WrapContent,
+				Padding:   layout.Inset{Right: values.MarginPadding5},
+				Direction: d.SelectedItemDirectionAfterCollapse,
+			}
+
+			return ll.Layout2(gtx, func(gtx C) D {
+				lbl := d.theme.Body2(item.Text)
+				if !d.expanded && len(item.Text) > maxDropdownItemTextLen {
+					lbl.Text = item.Text[:maxDropdownItemTextLen-3 /* subtract space for the ellipsis */] + "..."
+				}
+				lbl.Font.Weight = d.FontWeight
+				return lbl.Layout(gtx)
+			})
+		}),
+		layout.Rigid(func(gtx C) D {
+			return d.layoutActiveIcon(gtx, itemIndex)
+		}),
+	)
 }
 
-func (d *DropDown) layoutOption(gtx layout.Context, itemIndex int) D {
+func (d *DropDown) layoutExpandedItem(gtx C, itemIndex int) D {
 	item := d.items[itemIndex]
 	radius := Radius(0)
 	clickable := item.clickable
-	if !d.isOpen {
-		radius = Radius(8)
-		clickable = d.clickable
-	}
 
 	padding := values.MarginPadding10
 	if item.Icon != nil {
 		padding = values.MarginPadding8
 	}
 
-	dropdownWidth := d.Width
-	dropdownItemWidth := dropdownWidth
-	if dropdownWidth <= 0 {
-		dropdownWidth = defaultDropdownWidth(gtx, d.revs)
-	}
-
 	return LinearLayout{
-		Width:     dropdownWidth,
+		Width:     MatchParent,
 		Height:    WrapContent,
 		Clickable: clickable,
 		Padding:   layout.UniformInset(padding),
 		Border:    Border{Radius: radius},
 	}.Layout(gtx,
 		layout.Rigid(func(gtx C) D {
 			if item.Icon == nil {
-				return layout.Dimensions{}
+				return D{}
 			}
 
-			dropdownItemWidth -= gtx.Dp(values.MarginPadding24) // account for the dropdown Icon
 			return item.Icon.Layout24dp(gtx)
 		}),
-		layout.Rigid(func(gtx C) D {
-			gtx.Constraints.Max.X = dropdownItemWidth - gtx.Dp(values.MarginPadding50) // give some space for the dropdown Icon
-			gtx.Constraints.Min.X = gtx.Constraints.Max.X
-			return layout.Inset{
-				Right: unit.Dp(5),
-				Left:  unit.Dp(5),
-			}.Layout(gtx, func(gtx C) D {
+		layout.Flexed(1, func(gtx C) D {
+			if item.DisplayFn != nil {
+				return item.DisplayFn(gtx)
+			}
+
+			ll := LinearLayout{
+				Width:     MatchParent,
+				Height:    WrapContent,
+				Padding:   layout.Inset{Right: values.MarginPadding5},
+				Direction: layout.W,
+			}
+
+			return ll.Layout2(gtx, func(gtx C) D {
 				lbl := d.theme.Body2(item.Text)
-				if !d.isOpen && len(item.Text) > 14 {
-					lbl.Text = item.Text[:14] + "..."
+				if !d.expanded && len(item.Text) > maxDropdownItemTextLen {
+					lbl.Text = item.Text[:maxDropdownItemTextLen-3 /* subtract space for the ellipsis */] + "..."
 				}
+				lbl.Font.Weight = d.FontWeight
 				return lbl.Layout(gtx)
 			})
```

Comment:
This and the codes in Line 243 are similar... Why not move this into a wrapper method and pass the label text to it, rather than repeating it.